lib/orderbook/MatchingEngine.ts

```diff
@@ -66,34 +101,139 @@ class MatchingEngine {
+    const callback = (order: StampedOrder) => (order as StampedPeerOrder).peerPubKey === peerPubKey;
+
+    // remove from queues
+    const removedOrders = [
+      ...this.queues.buy.removeMany(callback),
+      ...this.queues.sell.removeMany(callback),
+    ] as StampedPeerOrder[];
+
+    // remove from lists. for further optimization, we can maintain a separate list for each peer pubKey
+    removedOrders.forEach((order: StampedPeerOrder) => {
+      const list = order.quantity > 0 ? this.peerOrders.buy : this.peerOrders.sell;
+      list.delete(order.id);
+    });
+
+    return removedOrders;
+  }
+
+  public removeOwnOrder = (orderId: string): StampedOwnOrder | undefined => {
+    const order = this.ownOrders.buy.get(orderId) || this.ownOrders.sell.get(orderId);
+    if (!order) {
+      return;
+    }
+
+    this.removeOrder(order, this.ownOrders);
+    return order;
+  }
+
+  private removePeerOrder = (order: StampedPeerOrder) => {
+    this.removeOrder(order, this.peerOrders);
+  }
+
+  private removeOrder = (order: StampedOrder, lists: OrderSidesLists<StampedOrder>) => {
+    const isBuyOrder = order.quantity > 0;
+    const list = isBuyOrder ? lists.buy : lists.sell;
+    const queue = isBuyOrder ? this.queues.buy : this.queues.sell;
+
+    list.delete(order.id);
+    queue.remove(order);
+  }
+
+  private getOrderList = (order: StampedOrder): OrderList<StampedOrder> => {
+    if (isOwnOrder(order)) {
+      return order.quantity > 0 ? this.ownOrders.buy : this.ownOrders.sell;
+    } else {
+      return order.quantity > 0 ? this.peerOrders.buy : this.peerOrders.sell;
+    }
+  }
+
+  private getOrders = <T extends orders.StampedOrder>(lists: OrderSidesLists<T>, limit?: number): OrderSidesArrays<T> => {
+    return {
+      buy: Array.from(lists.buy.values()).slice(0, limit || lists.buy.size),
+      sell: Array.from(lists.sell.values()).slice(0, limit || lists.sell.size),
+    };
+  }
+
+  public getPeerOrders = (limit?: number): OrderSidesArrays<StampedPeerOrder> => {
+    return this.getOrders(this.peerOrders, limit);
+  }
+
+  public getOwnOrders = (limit?: number) => {
+    return this.getOrders(this.ownOrders, limit);
+  }
+
   /**
    * Match an order against its opposite queue.
    * @returns a [[MatchingResult]] with the matches as well as the remaining, unmatched portion of the order
    */
-  public match = (takerOrder: StampedOwnOrder): matchingEngine.MatchingResult => {
+  private match = (takerOrder: StampedOwnOrder): matchingEngine.MatchingResult => {
     const isBuyOrder = takerOrder.quantity > 0;
     const matches: matchingEngine.OrderMatch[] = [];
     /** The unmatched remaining taker order, if there is still leftover quantity after matching is complete it will enter the queue. */
     let remainingOrder: StampedOwnOrder | undefined = { ...takerOrder };
 
-    const matchAgainst = isBuyOrder ? this.sellOrders : this.buyOrders;
+    const matchAgainst = isBuyOrder ? this.queues.sell : this.queues.buy;
```

Comment:
Maybe rename this to `queue`? I was never a big fan of `matchAgainst`. You can throw in a comment like `/** The queue to match the taker order against **/`.